### ISSUE TITLE
Server side upload

### ISSUE THREAD

**Issue by aadamsx** [CulturalMe/meteor-slingshot#164]
I'm creating a PDF file server side, and would like to do a Slingshot.Upload but I get an error when trying.  As a side note I can do this just fine on the client and can call Slingshot.createDirective just fine on the server side.

I'd like to do something like this server side:

```
var files = Meteor.settings.directory_path + userId + "/pdf-file.pdf";
  var uploader = new Slingshot.Upload("dropbox"); <-- !! I get an error here !!
  uploader.send(file, function (error, downloadUrl) {
    if (error) {
      console.error('Error uploading', uploader.xhr.response);
      alert (error);
    }
    else {
      var ownerId = userId; 
      var addedById = this.userId;
      var name = uploader.file.name;
      var type = uploader.file.type;
      var size = uploader.file.size;

      var fullPath = downloadUrl;
      var directory = userId;

      var indexOfDirectory = fullPath.indexOf(directory);
      var indexOfStringLength = fullPath.length;
      var key = fullPath.slice(indexOfDirectory, indexOfStringLength);
   }
});
```

**Comment by aadamsx** [comment]
It seems I might need to use Slingshot.S3Storage.upload on the server side, does anyone an example with parameters?

**Comment by aadamsx** [comment]
I've almost got things working using AWS.S3() directly instead.  I don't understand if Slingshot.S3Storage.upload is the right server API to use.

Is anyone monitoring this package anymore or is abandoned?  If it is, please designate someone else to take over, I see a lot of PRs just standing idle man.

**Comment by jaredtmartin** [comment]
I'm trying to do the same thing.  Have you had any success?

**Comment by bmustata** [comment]
Any update on this issue?

**Comment by acomito** [comment]
also trying to do this

**Comment by StorytellerCZ** [comment]
@aadamsx Did you found a solution?

**Comment by shivang007** [comment]
Any progress on this one?

**Comment by formspoint** [comment]
Indeed, I got his working a few years ago.  If you need help, PM me.

**Comment by shivang007** [comment]
@formspoint Thank you! But I figured out, instead of modifying this, using the AWS SDK's upload method would be a simple and clean solution.
And for anyone in the future stuck at figuring out "How to upload from server using slingshot", please avoid the modification and check this out:
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property
